src/shared/sleep-config.c

```diff
@@ -237,6 +298,24 @@ int find_hibernate_location(char **device, char **type, uint64_t *size, uint64_t
                                 log_debug("Ignoring compressed RAM swap device '%s'.", swap->device);
                                 continue;
                         }
+
+                        /* Check if swap matches the resume= value if configured. */
+                        if (!streq(resume, "0:0")) {
+                                _cleanup_free_ char *swap_partition;
```

Comment:
Same here. This will crash if the function returns before swap_partition has been intiialized. But you don't need this, use a static buffer instead and xsprintf.